lib/pirates/game_server/manager.ex

```diff
@@ -21,36 +21,56 @@ defmodule Pirates.GameServer.Manager do
         GenServer.call(@name, :next)
     end
 
+    def get_or_create_instance(name) do
+        GenServer.call(@name, {:get, name})
+    end
+
   ####################
   # Server Callbacks #
   ####################
 
   def init(:ok) do
-      servers = MapSet.new()
+      servers = Map.new()
       {:ok, servers}
   end
 
   def handle_call(:next, _from, servers) do
-    server = 
+      Logger.info("yo!")
+    {name, _} = 
         servers 
-        |> Enum.filter(fn (s) -> count(s) < @max_players_per_server end) 
-        |> Enum.min_by(&count/1, fn -> create_server() end)
-    {:reply, {:ok, server}, MapSet.put(servers, server)}
+        |> Enum.filter(fn ({_, i}) -> count(i) < @max_players_per_server end) 
+        |> Enum.min_by(fn ({_, i}) -> count(i) end, fn -> {System.unique_integer([:positive]), nil} end)
```

Comment:
When all the servers are full, it seems a bit funky to hand out server names without actually creating them.

Say `@max_players_per_server` is `n`:
if `n` players all call `:next` at roughly the same time _before_ any of them call `:get`, then we'll create `n` servers where we should have only made 1

Reply:
Yeah that's a good point. Not super critical for now but we should definitely address this in a future sprint.